observatory-dags/observatory/dags/workflows/onix_workflow.py

```diff
@@ -0,0 +1,314 @@
+        """
+
+        self.dag_id = dag_id
+        if dag_id is None:
+            self.dag_id = make_dag_id(self.DAG_ID_PREFIX, org_name)
+
+        # Set the Bigquery client object
+        self.client = bq_client
+        if bq_client is None:
+            self.client = Client()
+
+        # GCP parameters
+        self.workflow_dataset_id = "onix_workflow"
+        self.project_id = gcp_project_id
+        self.onix_dataset_id = "onix"
+        self.dataset_location = "us"
+        self.dataset_description = "ONIX workflow tables"
+
+        # ONIX release info
+        table_name, onix_release_date_str = self.get_latest_table_id()
+        self.onix_table_id = table_name
+        self.release_date = pendulum.parse(onix_release_date_str)
+        self.bucket_name = gcp_bucket_name
+        self.blob_path = os.path.join("onix_workflow", onix_release_date_str)
+
+        # Prepare filesystem
+        self.local_table_dir = self.blob_path
+        self.workslookup_filename = os.path.join(self.local_table_dir, "onix_workid_isbn.jsonl.gz")
+        self.workslookup_errors_filename = os.path.join(self.local_table_dir, "onix_workid_isbn_errors.jsonl.gz")
+        self.worksfamilylookup_filename = os.path.join(self.local_table_dir, "onix_workfamilyid_isbn.jsonl.gz")
+
+        Path(".", self.local_table_dir).mkdir(exist_ok=True, parents=True)
+
+        # Work (family) aggregation results
+        self.works = list()
+        self.works_families = list()
+
+        # Initialise Telesecope base class
+        super().__init__(
+            dag_id=self.dag_id, start_date=start_date, schedule_interval=schedule_interval, catchup=catchup
+        )
+
+        # Wait on ONIX telescope to finish
+        self.add_sensor(telescope_sensor)
+
+        # Aggregate Works
+        self.add_task(self.create_bq_isbn13_workid_lookup_table)
+        self.add_task(self.upload_isbn13_workid_lookup)
+        self.add_task(self.bq_load_workid_lookup)
+        self.add_task(self.bq_load_workid_lookup_errors)
+
+        # Aggregate Work families
+        self.add_task(self.create_bq_isbn13_workfamilyid_lookup_table)
+        self.add_task(self.upload_isbn13_workfamilyid_lookup)
+        self.add_task(self.bq_load_workfamilyid_lookup)
+
+        # Cleanup tasks
+        self.add_task(self.cleanup)
+
+    def make_release(self, **kwargs) -> None:
+        """Not needed.
+
+        :param **kwargs: Unused.
+        """
+
+        return None
+
+    def get_latest_table_id(self) -> Tuple[str, str]:
+        """Get the newest table name from the onix dataset.
+
+        :return: Latest table name and date string.
+        """
+
+        sql = f"SELECT table_name from {self.project_id}.{self.onix_dataset_id}.INFORMATION_SCHEMA.TABLES ORDER BY table_name DESC LIMIT 1"
+        df = self.client.query(sql).to_dataframe()
+        table_name = df["table_name"][0]
+        date_str = table_name[-8:]  # Last 8 digits is the YYYYMMDD date string
+        return table_name, date_str
```

Comment:
There is already a function here that can be used to select the table shard dates for a particular table: https://github.com/The-Academic-Observatory/observatory-platform/blob/develop/observatory-dags/observatory/dags/workflows/doi.py#L50-L68

It could be moved into the telescope utilities and renamed into something that makes better sense. You can also specify an end date to search up to (so that you only include tables up to the release date of the workflow for instance, in the case where there might be future releases of a table, such as when back-filling).

I think you can assume that the ONIX tables will always be called `onix`, so that could just be hardcoded as a constant in the telescope.